### PR TITLE
add node 0.12 to travis

### LINKED ISSUE #414: Add node v0.12 to travis
Node has [released](http://blog.nodejs.org/2015/02/06/node-v0-12-0-stable/) v0.12, this should be included in the travis tests.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8582913-add-node-v0-12-to-travis?utm_campaign=plugin&utm_content=tracker%2F2451736&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F2451736&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by dduugg**
Resolves #414. Also includes some travis-related cleanup, cutting the Travis build time in half (to <1 minute under normal load).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,12 @@
 language: node_js
+sudo: false
```
Comment (dduugg) [inline .travis.yml]: A tweak to allow Travis to start up faster. More info [here](http://docs.travis-ci.com/user/workers/container-based-infrastructure/)

File: Makefile [hunk 2]
```diff
@@ -4,22 +4,18 @@ MONGO_CONNECTION?=mongodb://localhost/test_db
...
+    NODE_ENV=test \
+    ${MONGO_SETTINGS} \
+    istanbul cover ./node_modules/mocha/bin/_mocha --report lcovonly -- -vvv -R tap ${TESTS}
 
 test:
-	${MONGO_SETTINGS} \
+    ${MONGO_SETTINGS} \
     mocha --verbose -vvv -R tap ${TESTS}
 
-travis: test travis-cov
```
Comment (dduugg) [inline Makefile]: Redundant – it's not necessary to test independently of generating the coverage report.

File: package.json [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "postinstall": "node node_modules/bower/bin/bower install"
   },
   "engines": {
-    "node": "~0.10.0"
+    "node": "0.10.x"
```
Comment (dduugg) [inline package.json]: Equivalent, just switching to the prevailing style. I assume we want to do more thorough testing before adding or switching to `"0.12.x"`.

File: Makefile [hunk 4]
```diff
@@ -9,17 +9,12 @@ MONGO_SETTINGS=MONGO_CONNECTION=${MONGO_CONNECTION} \
...
-	cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && \
-	rm -rf ./coverage
-
+.PHONY: test
 test:
 	${MONGO_SETTINGS} \
-    mocha --verbose -vvv -R tap ${TESTS}
+	mocha --verbose -vvv -R tap ${TESTS}
 
-travis: test travis-cov
```
Comment (dduugg) [inline Makefile]: Redundant – it's not necessary to separate testing and generating the coverage report.

File: package.json [hunk 5]
```diff
@@ -46,7 +46,6 @@
     "git-rev": "git://github.com/bewest/git-rev.git"
   },
   "devDependencies": {
-    "coveralls": "~2.11.2",
```
Comment (dduugg) [inline package.json]: This is now handled in the `after_script` command in `.travis.yml`.

File: .travis.yml [hunk 6]
```diff
@@ -1,14 +1,12 @@
...
-matrix:
-   allow_failures:
-     - node_js: "0.11"
+  - "0.10"
+  - "0.12"
 services:
   - mongodb
 before_script:
   - sleep 10
   - echo mongo mongo_travis
```
Comment (dduugg) [inline .travis.yml]: Just curious – what is this supposed to be doing?

File: Makefile [hunk 7]
```diff
@@ -9,17 +9,12 @@ MONGO_SETTINGS=MONGO_CONNECTION=${MONGO_CONNECTION} \
 
 all: test
 
-travis-cov:
-	NODE_ENV=test \
-	${MONGO_SETTINGS} \
-	istanbul cover ./node_modules/mocha/bin/_mocha --report lcovonly -- -vvv -R tap ${TESTS} && \
-	cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && \
-	rm -rf ./coverage
```
Comment (dduugg) [inline Makefile]: This is now handled in the `after_script` command in `.travis.yml`.

File: package.json (hunk 5 above)
Comment (jasoncalabrese) [inline package.json]: Why remove the Dev dependency for coveralls and replace it with a `npm install` in `after_script`?

File: .travis.yml (hunk 6 above)
Comment (jasoncalabrese) [inline .travis.yml]: Maybe left over from testing the Travis setup? @bewest?

File: package.json (hunk 5 above)
Comment (dduugg) [inline package.json]: Coveralls as I understand it is just a service to comment on github PRs and provide coverage badges on READMEs. What else are you looking for? You'll still get the coverage information when you run `make travis` or `instanbul cover...` locally.

See the `.travis.yml` file in the Express repo for an [example](https://github.com/strongloop/express/blob/master/.travis.yml) of this approach. (Note that Express installs a specific version of coveralls while I prefer to always install the latest. I'm open to dissent on this approach.)

Comment (jasoncalabrese) [inline package.json]: Makes sense, no reason to run it locally, only needs to run with Travis

File: Makefile (hunk 7 above)
Comment (bewest) [inline Makefile]: I'd keep all the bash in the `Makefile` and eliminate it from travis config.

**Comment by jasoncalabrese** [comment]
Need a new PR that targets dev too, since we're only updating master with releases

**Comment by dduugg** [comment]
Abandoning to target dev branch

**Pull request closed without merging**